### ISSUE TITLE
fix: stable2412

### ISSUE THREAD

**Issue by Daanvdplas** [paritytech/psvm#30]
When using `psvm -v "polkadot-stable2412"` the dependencies are downgraded.

**Comment by iulianbarbu** [comment]
I am noticing something similar when trying to get the corresponding versions for stable2412 for `polkadot-sdk/templates`, when running this action: https://github.com/paritytech/polkadot-sdk/actions/runs/12466779475/workflow. cc @ggwpez

**Comment by ggwpez** [comment]
To what are they downgraded? Can you post a diff please or send me a commit?

**Comment by ggwpez** [comment]
Issue seems to be that the 2412 release does not contain a `plan.toml` as usual and its `Cargo.lock` was not updated... pinged internally: https://github.com/paritytech/polkadot-sdk/tree/polkadot-stable2412

**Comment by Daanvdplas** [comment]
https://github.com/paritytech/polkadot-sdk/issues/6982

**Comment by iulianbarbu** [comment]
There is a Plan.toml on stable2412 branch right now, but `psvm` can only reference releases by their tag (e.g `polkadot-stable2412`), but this isn't enough because the Plan.toml can be frequently pushed to the branch later than the release tag. 

This means that if you want to use PSVM to update your dependencies according to a polkadot-sdk stable version tag, you can't tell if the existing release tags are enough, given the Plan.toml can be pushed before the tag but it can also be pushed after (as in the scope of this issue).

I am thinking about a few ways to unblock calling PSVM for a release that doesn't have the Plan.toml before its tag, or the Plan.toml gets updated:

1) No-op, wait for the patch release to come and run psvm against the patch release tag, which should contain at least the previous release Plan.toml. This option involves waiting for the patch release which happens once a month.

2) enable PSVM to use the `Plan.toml` pushed after a stable release tag, by giving it a commit (we can also ensure the commit lives on the release branch we're trying to update it for). This gives total freedom in choosing the Plan.toml to update from, as long as it is on the stable release branch.

3) Update each release page with a note about the commit where its Plan.toml can be found. The release page (and the associated release tag) can appear before the Plan.toml was pushed and a mention for its commit could be mentioned  in the release page, but even in this situation we can still communicate that the Plan.toml is pending, and PSVM can search for such text and return accordingly, instead of updating things in a wrongful manner.

4) Additionally to 3 we can mention tags in the release page, for a certain release Plan.toml. If the Plan.toml gets updated, we can tag the new plan commit by adding a suffix to the previous plan tag, like the patch suffix used with stable release patch versions.

**Comment by kianenigma** [comment]
For the case of templates, ideally they should be moved to the polkadot-sdk umbrella crate to no longer be at the mercy of `psvm`. 

Otherwise this issue seems quite important and makes `stable2412` not useable.

**Comment by ggwpez** [comment]
>I am thinking about a few ways to unblock calling PSVM for a release that doesn't have the Plan.toml before its tag, or the Plan.toml gets updated:

**Comment by ggwpez** [comment]
>I am thinking about a few ways to unblock calling PSVM for a release that doesn't have the Plan.toml before its tag, or the Plan.toml gets updated:

We should prod the release team to push this back up ASAP. A release is not done without updated Cargo.lock and Plan.toml.

>There is a Plan.toml on stable2412 branch right now, but psvm can only reference releases by their tag (e.g polkadot-stable2412), but this isn't enough because the Plan.toml can be frequently pushed to the branch later than the release tag.

So it seems that the release tag is not including the plan.toml? Then that is an issue. We should not try to work around those but report them in the release coordination chat. In the meantime i think it is possible to just also put a branch name into psvm instead of a tag? So you can put in the `stable2412` branch?
